sdk/metricsadvisor/ai-metrics-advisor/test/public/dataSourceCred.spec.ts

```diff
@@ -0,0 +1,287 @@
+        const updated = await client.getDatasourceCredential(createdServicePrincipalInKVCredId);
+        assert.ok(updated.id, "Expecting valid datasource credential");
+        assert.equal(updated.description, servicePrincipalInKVCredentialPatch.description);
+        assert.equal(updated.type, servicePrincipalInKVCredentialPatch.type);
+        assert.equal(updated.name, servicePrincipalInKVCredentialPatch.name);
+        assert.equal(
+          (updated as ServicePrincipalInKeyVaultDatasourceCredential).tenantId,
+          servicePrincipalInKVCredentialPatch.tenantId
+        );
+        assert.equal(
+          (updated as ServicePrincipalInKeyVaultDatasourceCredential).keyVaultClientId,
+          servicePrincipalInKVCredentialPatch.keyVaultClientId
+        );
+        assert.equal(
+          (updated as ServicePrincipalInKeyVaultDatasourceCredential).servicePrincipalIdNameInKV,
+          servicePrincipalInKVCredentialPatch.servicePrincipalIdNameInKV
+        );
+      });
+
+      it("lists datasource credentials one by one and by pages", async function() {
+        const iterator = client.listDatasourceCredential();
+        let result = await iterator.next();
+
+        assert.ok(result.value.id, "Expecting first datasource credential");
+        result = await iterator.next();
+        assert.ok(result.value.id, "Expecting second datasource credential");
+
+        const pageIterator = client.listDatasourceCredential().byPage({ maxPageSize: 2 });
+        let pageResult = await pageIterator.next();
+        assert.equal(pageResult.value.length, 2, "Expecting two entries in first page");
+        pageResult = await pageIterator.next();
+        assert.equal(pageResult.value.length, 2, "Expecting two entries in second page");
+      });
+
+      it("deletes sqlserver datasource credential", async function(this: Context) {
+        if (!createdSqlServerCredId) {
+          this.skip();
+        }
+        await verifyDatasourceCredentialDeletion(client, createdSqlServerCredId);
+      });
+
+      it("deletes datalake gen2 shared key datasource credential", async function(this: Context) {
+        if (!createdDatalakeCredId) {
+          this.skip();
+        }
+        await verifyDatasourceCredentialDeletion(client, createdDatalakeCredId);
+      });
+
+      it("deletes service principal datasource credential", async function(this: Context) {
+        if (!createdServicePrincipalCredId) {
+          this.skip();
+        }
+        await verifyDatasourceCredentialDeletion(client, createdServicePrincipalCredId);
+      });
+
+      it("deletes service principal in KeyVault datasource credential", async function(this: Context) {
+        if (!createdServicePrincipalInKVCredId) {
+          this.skip();
+        }
+        await verifyDatasourceCredentialDeletion(client, createdServicePrincipalInKVCredId);
+      });
+    });
+  })
+  .timeout(60000);
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```

Comment:
Do we disable similar things in other libs?

Reply:
Will need to investigate this. Will do for GA.